compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -308,14 +305,22 @@ private void validateEntityAnnotation(SyntaxNodeAnalysisContext ctx, TypeDefinit
                                 if (!name.equals(Constants.TABLE_NAME)) {
                                     validateEntityProperties(ctx, expressionNode.get(), tableFields);
                                 } else {
-                                    tableName = Utils.eliminateDoubleQuotes(expressionNode.get().toSourceCode().trim());
-                                    if (tableNames.contains(tableName)) {
-                                        reportDiagnosticInfo(ctx, mappingFieldNode.location(),
+                                    ExpressionNode exp = expressionNode.get();
+                                    if (exp instanceof BasicLiteralNode) {
```

Comment:
This validation needs to be done for the array itself as well as individual elements

Reply:
Already added this validation array as well https://github.com/ballerina-platform/module-ballerina-persist/pull/73/files#diff-2fef4b0b023d4af527c920976d479032ef537155e9422e1cfd555c006d0a6d96R374